doc/dev_guide/c_style_guide.rst

```diff
@@ -2,124 +2,36 @@
-please be warned: Launchpad silently deletes posts from non-subscribed members,
-thus please be sure to have subscribed to the list prior to posting. Additionally,
-some engineers are always present on #tarantool channel on irc.freenode.net.
-
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-          Commenting style
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-
-Use Doxygen comment format, Javadoc flavor, i.e. `@tag` rather than `\\tag`.
-The main tags in use are @param, @retval, @return, @see, @note and @todo.
-
-Every function, except perhaps a very short and obvious one, should have a
-comment. A sample function comment may look like below:
-
-.. code-block:: c
-
-    /** Write all data to a descriptor.
-     *
-     * This function is equivalent to 'write', except it would ensure
-     * that all data is written to the file unless a non-ignorable
-     * error occurs.
-     *
-     * @retval 0  Success
-     *
-     * @retval  1  An error occurred (not EINTR)
-     * /
-    static int
-    write_all(int fd, void \*data, size_t len);
-
-Public structures and important structure members should be commented as well.
-
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-           Header files
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-
-Use header guards. Put the header guard in the first line in the header,
-before the copyright or declarations. Use all-uppercase name for the header
-guard. Derive the header guard name from the file name, and append _INCLUDED
-to get a macro name. For example, core/log_io.h -> CORE_LOG_IO_H_INCLUDED. In
-``.c`` (implementation) file, include the respective declaration header before all
-other headers, to ensure that the header is self- sufficient. Header "header.h"
-is self-sufficient if the following compiles without errors:
-
-.. code-block:: c
-
-    #include "header.h"
-
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-          Allocating memory
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-
-Prefer the supplied slab (salloc) and pool (palloc) allocators to malloc()/free()
-for any performance-intensive or large  memory allocations. Repetitive use of
-malloc()/free() can lead to memory fragmentation and should therefore be avoided.
-
-Always free all allocated memory, even allocated  at start-up. We aim at being
-valgrind leak-check clean, and in most cases it's just as easy to free() the
-allocated memory as it is to write a valgrind suppression. Freeing all allocated
-memory is also dynamic-load friendly: assuming a plug-in can be dynamically loaded
-and unloaded multiple times, reload should not lead to a memory leak.
-
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-         Function naming
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-
-Our convention is to use:
+`developer discussion list <https://groups.google.com/forum/#!forum/tarantool>`_
+or on `StackOverflow <https://stackoverflow.com/questions/tagged/tarantool>`_.
+Additionally, some engineers are always present on #tarantool channel on
```

Comment:
I know that it is not practiced much in the doc team, but usually we try to make changes atomic. Here I see changes not only about C style, which makes it hard to review the C style updates. The diff is simply too big.
If the doc team does not care about atomicity, then nevermind. Otherwise I would split these non-style changes into a different commit.